components/ai_chat/resources/page/components/conversation_list/index.tsx

```diff
@@ -14,12 +14,36 @@ import DataContext from '../../state/context'
 import ContextMenuAssistant from '../context_menu_assistant'
 import { getLocale } from '$web-common/locale'
 import SiteTitle from '../site_title'
+import CodeBlock from '../code_block'
 
 const SUGGESTION_STATUS_SHOW_BUTTON: mojom.SuggestionGenerationStatus[] = [
   mojom.SuggestionGenerationStatus.CanGenerate,
   mojom.SuggestionGenerationStatus.IsGenerating
 ]
 
+function buildMessage(str: string) {
+  const parts: any = []
+
+  for (const match of str.matchAll(/```([^\n`]+)?\n?([\s\S]*?)```|`(.*?)`|([^`]+)/gs)) {
```

Comment:
Note that this will start/end fenced code blocks even when the backticks don't start at the beginning of a line (not necessarily an issue). Overall, the regex seems OK.

Also note that LLMs are happy outputting broken Markdown. We had some issues with Brave Search CodeLLM.

![image](https://github.com/brave/brave-core/assets/5838651/0d819473-567b-478a-9011-0722f9ce395f)